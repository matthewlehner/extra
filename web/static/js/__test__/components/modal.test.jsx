import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import Modal from "components/modal";

describe("Modal component", () => {
  it("renders properly", () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Modal onDismiss={jest.fn()} cancelPath="a string" />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders with children", () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Modal onDismiss={jest.fn()} cancelPath="a string">
          Iâ€™m a child element!
        </Modal>
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
