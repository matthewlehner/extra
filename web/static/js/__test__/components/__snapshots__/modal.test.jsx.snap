// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders properly 1`] = `
<div
  className="modal__overlay"
  onClick={[Function]}
>
  <div
    className="modal__container"
  >
    <header
      className="modal__header"
    >
      <button
        className="linkButton modal__dismiss"
        onClick={[Function]}
      >
        <svg
          className="icon"
        >
          <use
            xlinkHref="/images/app.svg#close-icon"
          />
        </svg>
      </button>
    </header>
    <div
      className="modal__body"
    />
  </div>
</div>
`;

exports[`Modal component renders with children 1`] = `
<div
  className="modal__overlay"
  onClick={[Function]}
>
  <div
    className="modal__container"
  >
    <header
      className="modal__header"
    >
      <button
        className="linkButton modal__dismiss"
        onClick={[Function]}
      >
        <svg
          className="icon"
        >
          <use
            xlinkHref="/images/app.svg#close-icon"
          />
        </svg>
      </button>
    </header>
    <div
      className="modal__body"
    >
      Iâ€™m a child element!
    </div>
  </div>
</div>
`;
